src/image.c

```diff
@@ -1903,21 +1903,21 @@ GdipGetImagePalette (GpImage *image, ColorPalette *palette, INT size)
 	int palette_entries;
 	int bytes_needed;
 
-        if (!image || !palette)
+	if (!image || !palette)
 		return InvalidParameter;
 
 	/* GDI+ doesn't support this for metafiles */
 	if (image->type != ImageTypeBitmap)
 		return NotImplemented;
 
 	if (!image->active_bitmap->palette) {
-		/* ColorPalette definition includes one ARGB member (ARGB Entries[1]). In reality there are 
-		 * Count entries allocated, where Count can be 0 (and required to substract the ARGB size)
-		 */
-		/* coverity[buffer_alloc] */
-		image->active_bitmap->palette = (ColorPalette*) GdipAlloc (sizeof(ColorPalette) - sizeof(ARGB));
-		image->active_bitmap->palette->Flags = 0;
-		image->active_bitmap->palette->Count = 0;
+		if (size >= 0 && size < sizeof(ColorPalette))
```

Comment:
so passing `size` smaller than zero is an accepted input or do we need an additional check?

Reply:
The following test passes against Windows GDI+. Negative sizes seem to be ignored.
```

	// Empty palette - negative size.
	palette->Count = 100;
	status = GdipGetImagePalette (bitmapImage, palette, -1);
	assertEqualInt (status, Ok);
	assertEqualInt (palette->Count, 0);
```